mobify/platform-scaffold
# [Checkout Schema] Split out shipping method

This PR moves the selected shipping method ID to be a child of the checkout state. It also renames the shipping method input to match out naming conventions (ie. no underscores).


 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-69
 **Linked PRs**: n/a

## Changes
- renamed shipping method input
- added receiveSelectedShippingMethod to checkout reducer
- moved selected shipping method if out of the shippingAddress branch and into the checkout branch

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- add an item to the cart
- proceed through the checkout to the shipping page
- fill out the address form
- select a shipping method other than the default and submit the page
- check that the shipping method id is a child of the checkout branch in the redux store (and that no shipping method id is stored in shippingAddress)
- navigate back to the shipping page and confirm that the correct shipping method is still selected
- Preview the [SFCC Site](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- add an item to the cart
- proceed through the checkout to the shipping page
- fill out the address form
- select a shipping method other than the default and submit the page
- check that the shipping method id is a child of the checkout branch in the redux store (and that no shipping method id is stored in shippingAddress)
- navigate back to the shipping page and confirm that the correct shipping method is still selected
